core/src/indexer/dependency.rs

```diff
@@ -275,22 +266,48 @@ impl ContractEventDependencies {
                                 })
                                 .collect(),
                             then: None,
-                        })),
-                    };
+                        };
 
-                    let dependency_tree =
-                        EventsDependencyTree::from_dependency_event_tree(&dependency_event_tree);
+                        dependency_event_tree
+                    });
 
-                    EventDependencies {
-                        tree: Arc::new(dependency_tree),
-                        dependency_events: dependency_event_tree.collect_dependency_events(),
-                    }
-                });
+                    let factory_dependency_event_tree = DependencyEventTree {
+                        contract_events: vec![ContractEventMapping {
+                            contract_name: factory.name,
+                            event_name: factory.event_name,
+                        }],
+                        then: Some(Box::from(dependency_event_tree)),
+                    };
 
-                dependencies.push(ContractEventDependencies {
-                    contract_name: contract.name.clone(),
-                    event_dependencies,
-                });
+                    let factory_dependency_tree = EventsDependencyTree::from_dependency_event_tree(
+                        &factory_dependency_event_tree,
+                    );
+
+                    dependencies.push(ContractEventDependencies {
+                        contract_name: contract.name.clone(),
+                        event_dependencies: EventDependencies {
+                            tree: Arc::new(factory_dependency_tree),
+                            dependency_events: factory_dependency_event_tree
+                                .collect_dependency_events(),
+                        },
+                    });
+                }
+                None => {
+                    if let Some(dependency_event_tree) = contract_dependencies_tree {
+                        let dependency_tree = EventsDependencyTree::from_dependency_event_tree(
+                            &dependency_event_tree,
+                        );
+
+                        dependencies.push(ContractEventDependencies {
```

Comment:
this was missing after refactoring, we need to populate dependencies array in case factory filter is not set